src/UserAction/StepExecutorTrait.php

```diff
@@ -102,18 +105,53 @@ protected function setFormField(Form $form, array $fields, string $step): Form
         foreach ($fields as $k => $val) {
             $form->getControl($k)->set($val);
         }
+
         $this->hook(self::HOOK_STEP, [$step, $form]);
 
         return $form;
     }
 
+    /**
+     * Set model for userAction arguments.
+     * Override existing argument with model definition.
+     */
+    protected function getModelForArgs(): Model
+    {
+        $args = $this->getAction()->args;
+        if (array_key_exists('__atk_model', $args)) {
+            /** @var Model $argsModel */
+            $argsModel = Factory::factory($args['__atk_model']);
+            // if seed is supplied, we need to initialize.
+            if (!$argsModel->_initialized) {
+                $argsModel->invokeInit();
+            }
+
+            unset($args['__atk_model']);
+        } else {
+            $argsModel = new Model(new Array_([]));
+        }
+
+        foreach ($args as $key => $val) {
+            $argsModel->addField($key, $val);
```

Comment:
why args are not a Model?

Reply:
UserAction->args are not model, they never was. Only now executor will use them as a model.